spec/rubocop/cop/metrics/single_line_complexity_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Metrics::SingleLineComplexity, :config do
+  let(:cop_config) { { 'Max' => 2 } }
+
+  context 'single line code' do
+    it 'registers an offense when ABC per line > 2' do
+      expect_offense(<<~RUBY)
+        a ? b : c
+        ^^^^^^^^^ Assignment Branch Condition size too high for line 1. [3/2]
+      RUBY
+    end
```

Comment:
It seems off to me that converting to a long form `if` would not register an offense here, especially because `Metrics/AbcSize` would score the ternary as 3.16 but the `if` as 3.61. For that matter, why does this cop produce a different score?

```ruby
expect_no_offenses(<<~RUBY)
  if a
    b
  else
    c  
  end
RUBY
```

Reply:
This cop looks at the score within a single line. When you unwrap that statement into multiple line, there's no longer a single line with a score > 2. I can show you with some debugging code mapping lines to source to AST if you like.